packages/@webex/plugin-meetings/src/meeting/index.ts

```diff
@@ -4548,6 +4549,26 @@ export default class Meeting extends StatelessWebexPlugin {
 
     this.isMultistream = !!options.enableMultistream;
 
+    try {
+      // refresh the permission token if its about to expire in 10sec
+      await this.checkAndRefreshPermissionToken(
+        MEETING_PERMISSION_TOKEN_REFRESH_THRESHOLD_IN_SEC,
+        'ttl-join'
+      );
+    } catch (error) {
+      LoggerProxy.logger.error('Meeting:index#join --> Failed to refresh permission token:', error);
+
+      if (
+        error instanceof CaptchaError ||
+        error instanceof PasswordError ||
+        error instanceof PermissionError
+      ) {
+        // if refresh permission token requires captcha, password or permission, we are throwing the errors
+        // and bubble it up to Cantina
+        return Promise.reject(error);
```

Comment:
there is a lot of code in the catch() block starting at line 4587 that normally gets executed when something fails, but if `checkAndRefreshPermissionToken()` fails with one of captcha/password/permisison errors, then we fail and we don't execute that code. I think we need that to be executed.